### ISSUE TITLE
Reported errors depend on where ansible-lint is excecuted

### ISSUE THREAD

**Issue by BonaBeavis** [ansible/ansible-lint#247]
# Issue Type
- Bug Report

# Ansible and Ansible Lint details

```
ansible 2.2.1.0
ansible-lint 3.4.11
```

- ansible installation method: pip
- ansible-lint installation method: pip

# Desired Behaviour

Please give some details of the feature being requested
or what should happen if providing a bug report

# Actual Behaviour (Bug report only)
```
➜  cis git:(master) ✗ ansible-lint ./roles/etcd
"No Errors"   
➜  cis git:(master) ✗ cd roles 
➜  roles git:(master) ✗ ansible-lint ./etcd
[DEPRECATION WARNING]: docker is kept for backwards compatibility but usage is discouraged. The module documentation details page may explain more about this rationale..

This feature will be removed in a future release. Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
```

# Desired Behavior
```
➜  cis git:(master) ✗ ansible-lint ./roles/etcd
[DEPRECATION WARNING]: docker is kept for backwards compatibility but usage is discouraged. The module documentation details page may explain more about this rationale..

This feature will be removed in a future release. Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
➜  cis git:(master) ✗ cd roles 
➜  roles git:(master) ✗ ansible-lint ./etcd
[DEPRECATION WARNING]: docker is kept for backwards compatibility but usage is discouraged. The module documentation details page may explain more about this rationale..

This feature will be removed in a future release. Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
```

**Comment by BonarBeavis** [comment]
This works: `ansible-lint ./roles/etcd/`

**Comment by willthames** [comment]
I'm not quite sure what happened here - is it working as expected after all? What caused you to think it wasn't working previously?

**Comment by strangeman** [comment]
@willthames it not working without trailing slash.

```
str@strangebook:/mnt/src/sources/playbooks$ ansible-lint ./roles/basic-linux
# nothing
str@strangebook:/mnt/src/sources/playbooks$ ansible-lint ./roles/basic-linux/
[ANSIBLE0002] Trailing whitespace
/mnt/src/sources/united_trader/luna-playbooks/roles/goozbach.epel/meta/main.yml:4
  description: EPEL repository 

[ANSIBLE0002] Trailing whitespace
/mnt/src/sources/united_trader/luna-playbooks/roles/goozbach.epel/tasks/main.yml:60
```

**Comment by willthames** [comment]
ok, I just didn't understand why this was closed.

**Comment by willthames** [comment]
Possibly related to #210 and #193 (any fix needs to check for regressions related to those)

**Comment by 5nafu** [comment]
I just found this issue and unfortunately this seems still to be an issue in Version 3.5.0
My directory structure is:
```$ tree -d /a/long/test/path/ansible-common/
/a/long/test/path/ansible-common/
└── common
    ├── defaults
    ├── files
    ├── handlers
    ├── meta
    ├── tasks
    ├── templates
    └── vars
```
Where ansible-common is the git root.
### Expected behaviour
Ansible-lint should output the same messages, independent of the path to the directory.

### Actual behaviour
* When running from inside the role directory: :heavy_check_mark: 
   ```bash
   /a/long/test/path/ansible-common/common $ ansible-lint .
   [403] Package installs should not use latest
   /a/long/test/path/ansible-common/common/tasks/basic_apps.yml:4
   Task/Handler: Install common software

   [403] Package installs should not use latest
   /a/long/test/path/ansible-common/common/tasks/basic_apps.yml:46
   Task/Handler: Install plattform specific packages

   [403] Package installs should not use latest
   /a/long/test/path/ansible-common/common/tasks/basic_apps.yml:65
   Task/Handler: Install plattform Debian<=8 specific security packages
   ```
* When running from above the git root directory: :heavy_check_mark: 
   ```bash
   /a/long/test/path $ ansible-lint ansible-common/common/
   [403] Package installs should not use latest
   /a/long/test/path/ansible-common/common/tasks/basic_apps.yml:4
   Task/Handler: Install common software

   [403] Package installs should not use latest
   /a/long/test/path/ansible-common/common/tasks/basic_apps.yml:46
   Task/Handler: Install plattform specific packages

   [403] Package installs should not use latest
   /a/long/test/path/ansible-common/common/tasks/basic_apps.yml:65
   Task/Handler: Install plattform Debian<=8 specific security packages
   ```
* When running from anywhere with a absolute path: :heavy_check_mark: 
   ```bash
   /a/long/test/path/ansible-common $ ansible-lint $(pwd)/common/
   [403] Package installs should not use latest
   /a/long/test/path/ansible-common/common/tasks/basic_apps.yml:4
   Task/Handler: Install common software

   [403] Package installs should not use latest
   /a/long/test/path/ansible-common/common/tasks/basic_apps.yml:46
   Task/Handler: Install plattform specific packages

   [403] Package installs should not use latest
   /a/long/test/path/ansible-common/common/tasks/basic_apps.yml:65
   Task/Handler: Install plattform Debian<=8 specific security packages
   ```
* When running from git root with only the role-directory: :heavy_exclamation_mark:
   ```
   ansible-common $ ansible-lint common/
   ansible-common $
   ```
* When running with a relative path not going to `/`: :heavy_exclamation_mark:
   ```
   /a/long/test/path//ansible-common $ ansible-lint ../ansible-common/common/
   /a/long/test/path//ansible-common $
   ```
* Now, the really weird stuff: When running with a relative path **going** to `/`: :heavy_check_mark: 
   ```
   /a/long/test/path/ansible-common $ ansible-lint ../ansible-common/common/
   /a/long/test/path/ansible-common $ ansible-lint ../../path/ansible-common/common/
   /a/long/test/path/ansible-common $ ansible-lint ../../../test/path/ansible-common/common/
   /a/long/test/path/ansible-common $ ansible-lint ../../../../long/test/path/ansible-common/common/
   /a/long/test/path/ansible-common $ ansible-lint ../../../../../a/long/test/path/ansible-common/common/
   [403] Package installs should not use latest
   /a/long/test/path/ansible-common/common/tasks/basic_apps.yml:4
   Task/Handler: Install common software
   
   [403] Package installs should not use latest
   /a/long/test/path/ansible-common/common/tasks/basic_apps.yml:46
   Task/Handler: Install plattform specific packages
   
   [403] Package installs should not use latest
   /a/long/test/path/ansible-common/common/tasks/basic_apps.yml:65
   Task/Handler: Install plattform Debian<=8 specific security packages
   ```

**Comment by awcrosby** [comment]
Error with not linting when having or not having trailing slash was resolved in #390
